Arduino/MiP_ProMini_Pack_Library/src/mip.cpp

```diff
@@ -2127,6 +2381,26 @@ void MiP::processOobResponseData(uint8_t commandByte)
     case MIP_CMD_GET_STATUS:
         length = 2;
         break;
+    case MIP_CMD_RECEIVE_IR_DONGLE_CODE:
```

Comment:
I thought I had added a comment here but it seems to have disappeared or I didn't save it.

I know that Serial.readBytes() will wait for 100 milliseconds and then timeout so this might be a better way to approach this code (I didn't try compiling so no doubt has typos):
```c++
        uint8_t nibbles[2];
        size_t bytesRead = Serial.readBytes(nibbles, sizeof(nibbles));
        if (bytesRead != sizeof(nibbles))
        {
            MiPStream.println(F("MiP: Missing IR code length"));
            return;
        }
        length = (parseHexDigit(nibbles[0]) << 4) | parseHexDigit(nibbles[1]);
        if (length < 2 || length > 4)
        {
            uint8_t discardedBytes = discardUnexpectedSerialData();
            MiPStream.print(F("MiP: Bad IR code length: "));
            MiPStream.print(length, HEX);
            MiPStream.print(F(" (discarded "));
            MiPStream.print(discardedBytes);
            MiPStream.println(F(" bytes)"));
            return;
        }
```

Reply:
That's very helpful but going to two-byte sends means we won't need to do all this.  I'll have to remember this for when we come back to implementing three- and four-byte sends.

Reply:
It won't be very robust if you don't read in the length bytes and just assume it contains 2-bytes of data.

Reply:
I still intend to read the length but it will happen below the switch statement with:

```
    // Read in the additional bytes of the notification.
    uint8_t buffer[2 * 2];
    size_t  bytesRead = Serial.readBytes(buffer, length * 2);
```

Like all the other read commands.  Buffer will need to be buffer[3 * 2] to read the length and two data bytes.

Reply:
Ok. I wouldn't do it that way but that's just me. I would write the receive code to work for 4-bytes as you currently have it and then just modify your send code to only send 2-bytes for now. What would your proposal above do if another MiP was firing shots via the iOS application?

Reply:
Ah, as always you’re right. I hadn’t thought of receiving code from outside of that which the ProMini would send. I’ll stay on course to a variable length receive. Heck, one less thing to worry about when it comes time to fix the variable length sends.